### DISCUSSION TITLE
custom scripts called by event rule

### DISCUSSION THREAD

**Discussion by Frank17489** [netbox-community/netbox#16952] [Q&A] [answered]
Hi,

I prepare a custom script, called by an event rule. The first script works. 
Data in script are a json structure - fine. 

But my second try dosn't work.

I want to find out how the data is structured / reverse engineering ;-) . For this I want to write the data in a textfile with **json.dumps(data)**

I got an errormessage (after event: create a device)

An exception occurred: `TypeError: Object of type __proxy__ is not JSON serializable.

Whats wrong and is there a better way to discover the datastructure for the different event I want to use.

~Frank

**Comment by PaulR282** [comment, marked as answer]
You can look at the data through "background tasks". If you are on v4, Admin -> Background Tasks and there search for your task after the script ran.

**Reply by Frank17489** [reply]
Ok, I found this, the data structure is a json, but json.dumps() gives the same error. But it's not necessary anymore, because I see the data structure in background tasks and can access the values (for example data['id'])
